### PR TITLE
Loading dbCore fails on Windows

### PR DESCRIPTION
**Pull request by hanak**
**Problem**
Using package on Windows (Windows 10) fails on loading dbCode library even though the library is built and available at the path (including its dependencies).

**Analysis**
The Python code assumes Linux environment in names of shared libraries and paths.

**Fix**
* Use correct Windows shared library names and path separators if Python detects Windows.
* Allow "EPICS_HOST_ARCH" environment variable in `examples/test.py`. This enables use of cross-compiled MingW binaries with no modifications `examples/test.py`.
* Use file output on Windows because `/dev/stdout` is not available on Windows.

**Testing**
* Run `python examples/test.py` on Windows 10 (32-bit binaries built with MingW)
* Run `python examples/test.py` on Debian 10  (64-bit binaries)

### CODE REVIEW

File: epicsdbbuilder/dbd.py [hunk 1]
```diff
@@ -108,13 +109,16 @@ def LoadDbdFile(dbdfile, on_use = None):
         os.chdir(dirname)
 
     # We add <epics_base>/dbd to the path so that dbd includes can be resolved.
+    separator = ':'
+    if platform.system() == 'Windows':
+        separator = ';'
+
     status = mydbstatic.dbReadDatabase(
-        ctypes.byref(_db), filename, '.:%s/dbd' % _epics_base, None)
+        ctypes.byref(_db), filename, separator.join(['.', os.path.join(_epics_base, 'dbd')]), None)
```
Comment (Araneidae) [inline epicsdbbuilder/dbd.py:117]: Please split this line to fit within 80 columns.  I suggest splitting after `filename,` and aligning `separator` with `ctypes` on the line above.

File: epicsdbbuilder/mydbstatic.py [hunk 2]
```diff
@@ -67,12 +67,19 @@ def ImportFunctions(epics_base, host_arch):
...
+    library_name_format = 'lib{}.so'
+    library_location = 'lib'
+    if platform.system() == 'Windows':
+        library_name_format = '{}.dll'
+        library_location = 'bin'
+
     try:
         libdb = PyDLL(os.path.join(
-            epics_base, 'lib', host_arch, 'libdbCore.so'))
+            epics_base, library_location, host_arch, library_name_format.format('dbCore')))
```
Comment (Araneidae) [inline epicsdbbuilder/mydbstatic.py:79]: Same as before: please split this line and line 82 below to fit in 80 columns.  Split after `host_arch,` and align with `epics_base,`.

File: examples/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+test_output.db
```
Comment (Araneidae) [inline examples/.gitignore:1]: There's a missing newline at the end of this file, please do add.

**Review by Araneidae — CHANGES_REQUESTED** [review]
Thank you for this pull request.  I had no idea that epicsdbbuilder would actually work on Windows!

This looks like a clean patch, but can you please add a further commit to fix the trivial layout issues, and then I'll happily pull.

**Comment by hanak** [comment]
Thank you for the review. I have updated the code accordingly.

**Comment by Araneidae** [comment]
Thank you for this change.

**Pull request merged**